1단계 - 웹 성능 테스트

OhSeungCheol opened the pull request:
안녕하세요.

1단계 - 웹 성능 테스트 미션 리뷰 부탁드립니다.

감사합니다.

OhSeungCheol committed:
docs: 웹 성능 예산 작성

OhSeungCheol commented:
'웹 성능 진단하기' 과정 작업 환경이 정확히 파악이 안되서 몇가지 질문 드립니다.

1. infra-subway-monitoring app 을 구동하는 위치는 AWS Public 용 EC2 인가요? 아니면 로컬에서 해도 상관이 없나요?
2. ./gradelw clean build 명령어를 실행하면 첨부 이미지처럼 지하철_노선_목록_응답됨 테스트가 실패하면서 종료됩니다($PROJECT/build/libs 아래 jar 파일이 생성은 됩니다.) 해당 명령어 실행 시 App 빌드 및 실행까지 되는게 정상동작인가요?
3. DB 는 저장소에 있는 docker compose 설정을 사용해서 띄우는게 맞나요? 데이터가 따로 없던데 빈 상태로 진행하는건지, 따로 데이터를 추가해야하는건지요.

설정을 잘못하고 있는건지 이해를 잘못하고 있는지.. 확인할 포인트가 있다면 답변 부탁드립니다.

![image](https://user-images.githubusercontent.com/20293663/206126002-6b133802-8a17-43fa-aec1-6b67b3970a9f.png)

brainbackdoor commented:
안녕하세요~ 미션을 진행하는데, 모호한 부분이 있었군요~
1. Public EC2 입니다. 성능테스트를 하는데, 로컬에서 하는건 의미가 없습니다. 강의에서도 이야기했지만, 성능테스트를 하는 대상은 운영환경과 유사해야 합니다.
2. 이 부분도 강의 때 말씀드렸는데요. Thread dump를 떠보지 않은 분들을 위해 '애플리케이션 진단하기' 강의자료 내에 실습을 두었어요. Thread dump를 이전에 경험해 본 분들이라면, 커밋 로그 보고 불필요한 코드 제거하고 진행하시면 됩니다.
3. docker compose 설정은 무시하시면 됩니다. 1주 차 미션과 동일하게 내부망 디비를 바라보도록 설정 구성 후 배포하시고 진행하시면 됩니다.

brainbackdoor reviewed (changes requested):
안녕하세요~ 질문주신 부분 피드백 남겨두었으니, 참고바래요 🙏🏻
좋은 하루 보내세요~

Commit:
docs: 정량 기준 지표 기준 수정

자바 스크립트 파일 크기 합 기준을 10 MB 라고 결정했었으나,
정량 기준 지표 샘플을 살펴보니 대부분 2 MB 언저리인 것을 확인.
10 MB 는 너무 크게 잡은 것 같아서 기준 변경 (기술적인 근거 X)

Commit:
docs: 유사 서비스 관련 지표 추가 및 서비스 목표 상세화

OhSeungCheol commented:
강의에 집중을 안했군요(웹 성능 테스트 미션 부분 까지만 봐서 그럴 수도 있습니다) 😅 
답변 확인했습니다. 감사합니다.
M1 내용 보완한 부분 다시 리뷰 요청 부탁리겠습니다.

brainbackdoor commented on README.md:
```diff
@@ -40,9 +40,34 @@ npm run dev
 
 2. 웹 성능예산을 바탕으로 현재 지하철 노선도 서비스의 서버 목표 응답시간 가설을 세워보세요.
 
+- 유사 서비스의 메인 페이지의 응답 시간은 1~6 초 사이에 이루어지고 있다.
+이 결과와 3초의 법칙을 근거로, 지하철 노선도 서비스의 메인 페이지의 응답 시간은 3초 이내가 되는 것을 목표로 한다.
+
+- 유사 서비스의 경로 탐색 응답 시간은 평균 0.5 초 내에 이루어진다. 
+(단, 카카오맵 서비스의 경우 가능한 모든 경로를 탐색하는 더 확장된 기능을 제공하기에 응답 시간이 길게 측정 되었다고 판단한다.)
+이 측정 결과의 20 % 를 초과하지 않는 0.6 초 미만을 지하철 노선도 서비스의 경로 탐색 응답 시간 목표로 한다.
```
👍🏻

brainbackdoor reviewed (approved):
1단계 미션 결과를 토대로 2단계 부하테스트도 직접 계측해보셔요~ 🙏🏻
1주일간 고생하셨어요~ 주말 잘 보내시고, 수업 때 뵈어요~
